sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -66,15 +68,32 @@ public LogsQueryClient(Uri endpoint, TokenCredential credential) : this(endpoint
         /// <param name="endpoint">The service endpoint to use.</param>
         /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
         /// <param name="options">The <see cref="LogsQueryClientOptions"/> instance to use as client configuration.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public LogsQueryClient(Uri endpoint, TokenCredential credential, LogsQueryClientOptions options)
         {
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
 
             Endpoint = endpoint;
             options ??= new LogsQueryClientOptions();
-            var authorizationScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? LogsQueryAudience.AzurePublicCloud : options.Audience)}";
-            authorizationScope += "//.default";
+
+            // Set authorization scope from Endpoint if Audience is not set.
+            string authorizationScope = "";
+            if (string.IsNullOrEmpty(options.Audience?.ToString()))
+            {
+                options.Audience = endpoint.AbsoluteUri;
+                // Endpoint.AbsoluteUri includes an extra / so only adding one here
+                authorizationScope = $"{options.Audience}/.default";
+            }
+            else if (endpoint.Host != new Uri(options.Audience.ToString()).Host)
+            {
+                throw new InvalidOperationException("The endpoint URI and audience do not match. If setting the Audience which is regionally specific, please use the LogsQueryClient(TokenCredential, LogsQueryClientOptions) constructor.");
```

Comment:
```suggestion
                throw new InvalidOperationException("The endpoint URI and audience do not match. If setting the Audience to a value that is regionally specific, please use the LogsQueryClient(TokenCredential, LogsQueryClientOptions) constructor.");
```

Reply:
Actually maybe this?

"The endpoint URI and audience do not match. If setting the Audience to a regionally specific value, please use the LogsQueryClient(TokenCredential, LogsQueryClientOptions) constructor.